[WIP] Adding what's new content to guide

bergerhoffer opened the pull request:
Modeling how it would look to add our what's new content to the actual guide.

Preview: http://file.rdu.redhat.com/~ahoffer/2022/main-whats-new.html#whats-new

mportman12 commented:
Style Council meeting outcome: Council agreed to including a What's New; don’t include past history from the last few years, just do most recent and go forward.

bergerhoffer commented:
@redhat-documentation/ccs-style-council This is ready for review. Per the last meeting, we won't include all of the past history, so I started fresh with 2023, with a link to the GitHub history page for all past years. We'll update every month to include the previous month's updates.

Preview: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new.html#whats-new

bburt-rh commented:
Lgtm, but I have what I admit is a nitpick. Is there any particular order to the items listed? Is the list chronological by when the PR merged? Should we consider doing it alphabetically? If you don't think it matters, then leave it as-is

bergerhoffer commented:
> Lgtm, but I have what I admit is a nitpick. Is there any particular order to the items listed? Is the list chronological by when the PR merged? Should we consider doing it alphabetically? If you don't think it matters, then leave it as-is

Yes, it's chronological by newest first (when the PR merged). I don't have a strong opinion on which way to order, other than that this is the way we've been doing it and it would take a bit of time to figure out how to reword if we put the term first, for example. I checked IBM and it doesn't look like their what's new section is necessarily alphabetized. But I'm open to changing it if others think we should.

mportman12 commented on supplementary_style_guide/introduction/whats-new.adoc:
```diff
@@ -0,0 +1,36 @@
+[[january-2023]]
+=== January 2023
+
+* Removed "through" entry in favor of following the _IBM Style_ guidance on number ranges
```
```suggestion
* Removed "through" entry in favor of following the _IBM Style_ guidance on number ranges.
```

bergerhoffer replied:
@mportman12 I do already link to all of the entries (or am I missing something)? Only "through" is not linked, because it was removed (so it doesn't have a place to link to anymore).

mportman12 reviewed (commented):
One small typo. Do we want to link each item to its entry below in the SSG? I personally think this would be a bit too much work for not enough reward, but throwing it out there in case others feel differently.

bburt-rh commented:
> Yes, it's chronological by newest first (when the PR merged). I don't have a strong opinion on which way to order, other than that this is the way we've been doing it and it would take a bit of time to figure out how to reword if we put the term first, for example. I checked IBM and it doesn't look like their what's new section is necessarily alphabetized. But I'm open to changing it if others think we should.

Ok. Thanks, that makes sense to keep it consistent with historic practices. I'm fine with leaving it as-is.

bredamc commented:
Given our target audience, I think it _would_ be better to list alphabetically and to link to current entries where applicable, but I understand that both of those RFEs would require extra work.

bergerhoffer commented:
Thank you both - let me see what I can come up with!

bergerhoffer commented:
@bburt-rh @bredamc I have a few options for you because I couldn't decide what I liked best :wink: I didn't update the source here yet, just uploaded previews of the options:

Option 1: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new-alphabetical-1.html#whats-new
Option 2: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new-alphabetical-2.html#whats-new
Option 3: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new-alphabetical-3.html#whats-new

I probably like option 2 the best, since it separates glossary and style entries. Option 3 is most like the ISG, but it might be a bit to drawn out and long for our needs. Let me know what you think!

p.s. I also removed the overarching year headings (compared to the original preview) and just added a note about where to find the old history. I did this because of option 3, which added another level heading, so they were getting too small with the year. If we pick option 1 or 2, I could add that back if we think it was better, or keep it as I have it now as just as note.

bburt-rh commented:
> @bburt-rh @bredamc I have a few options for you because I couldn't decide what I liked best wink I didn't update the source here yet, just uploaded previews of the options:
> 
> Option 1: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new-alphabetical-1.html#whats-new Option 2: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new-alphabetical-2.html#whats-new Option 3: http://file.rdu.redhat.com/~ahoffer/2023/main-whats-new-alphabetical-3.html#whats-new
> 
> I probably like option 2 the best, since it separates glossary and style entries. Option 3 is most like the ISG, but it might be a bit to drawn out and long for our needs. Let me know what you think!
> 
> p.s. I also removed the overarching year headings (compared to the original preview) and just added a note about where to find the old history. I did this because of option 3, which added another level heading, so they were getting too small with the year. If we pick option 1 or 2, I could add that back if we think it was better, or keep it as I have it now as just as note.

- I'm partial to 3 because it uses fewer words to achieve the same thing as 1 and 2. 2 would be my second choice.
- For option 3, consider removing the "Added guidance on" in the Style > Added section and replacing with "Guidance on..." It's redundant with the "added" section heading. Or is there some way to remove the description entirely and just including the link to make it more like the glossary list of links.
- For option 3, I also assume that we would have headings for "Changed" and "Removed" in this section, if appropriate.

bredamc commented:
I too prefer Option 3. I think it's fine to omit the ("Added", "Changed", "Removed") subheadings if we don't have any content for those sections. I wonder if the entire "What's New" section would be better placed in an appendix at the end of the guide, given that it's reference material -- and perhaps include a link to that appendix from the "How to use this guide" section? (I know the current location is the same as in IBM Style, just seems odd to me :))

mportman12 commented:
I like the sleek look of option 3, but we aren't likely to encounter a situation where style guidance is being removed, only ever added or updated. It's a bit of a misnomer to set up the Glossary entries and Style guidance sections as mirroring each other with "Added" and "Removed" designations. Therefore, I prefer option 2, which would dodge around the issue entirely.

bredamc commented:
I'd hope that we would be able to remove style guidance; for example, when _IBM Style_ is updated to account for Red Hat exceptions, or when Red Hat advice changes to align with _IBM Style_ :)

In any case, we shouldn't include the subheadings each time -- they should be included only when they are actually needed to describe a change.

bergerhoffer commented:
> > I probably like option 2 the best, since it separates glossary and style entries. Option 3 is most like the ISG, but it might be a bit to drawn out and long for our needs. Let me know what you think!
> > p.s. I also removed the overarching year headings (compared to the original preview) and just added a note about where to find the old history. I did this because of option 3, which added another level heading, so they were getting too small with the year. If we pick option 1 or 2, I could add that back if we think it was better, or keep it as I have it now as just as note.
> 
>     * I'm partial to 3 because it uses fewer words to achieve the same thing as 1 and 2. 2 would be my second choice.
> 
>     * For option 3, consider removing the "Added guidance on" in the Style > Added section and replacing with "Guidance on..." It's redundant with the "added" section heading. Or is there some way to remove the description entirely and just including the link to make it more like the glossary list of links. Maybe replace the heading with "Style guidance"
> 
>     * For option 3, I also assume that we would have headings for "Changed" and "Removed" in this section, if appropriate.

I don't like the idea of removing the description entirely for the style guidance section, because some context is necessary for some of them (especially when we remove things, so we can say why we removed it). Option 2 is more consistent between the two sections, and isn't repetitive with the "Added" part. Option 3 uses less words, but is longer vertically with the extra headings.

Looks like we have 2 for Option 2 and 2 for Option 3 :). @redhat-documentation/ccs-style-council Anyone else want to weigh in on which format they like better for the what's new page?

bergerhoffer commented:
I'd like to get this new section in prior to the email going out for February's meeting. I updated this PR to go with option 2 since it's straightforward and leaves flexibility for us to add additional context if needed.

@redhat-documentation/ccs-style-council This is ready for review.

mportman12 reviewed (approved):
LGTM

bburt-rh commented:
lgtm

dfitzmau commented:
LGTM. Merging.
